### PR TITLE
[TH2-3096] Migrated to codec 4.5.0-dev

### PR DESCRIPTION
**Pull request by Nikita-Smirnov-Exactpro**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -23,13 +23,16 @@ repositories {
...
+    implementation 'com.exactpro.th2:codec:4.5.0-dev-1783818520-SNAPSHOT'
 
-    implementation 'com.exactpro.th2:common:3.29.0'
+    implementation 'com.exactpro.th2:common:3.32.1'
 
     implementation 'org.quickfixj:quickfixj-core:2.3.1'
     implementation 'org.quickfixj:quickfixj-messages-all:2.3.1'
 
+    compileOnly 'com.google.auto.service:auto-service:1.0.1'
+    annotationProcessor 'com.google.auto.service:auto-service:1.0.1'
```
Comment (cordwelt) [inline build.gradle:40]: Please also remove `com.github.harbby.gradle.serviceloader` plugin

File: src/main/java/com/exactpro/th2/codec/qfj/QFJCodec.java [hunk 2]
```diff
@@ -31,19 +49,17 @@
 
 import static com.exactpro.th2.common.message.MessageUtils.toJson;
 
-
+@AutoService(IPipelineCodec.class)
```
Comment (cordwelt) [inline src/main/java/com/exactpro/th2/codec/qfj/QFJCodec.java:56]: Redundant - we only we need factory to be discoverable

File: src/main/java/com/exactpro/th2/codec/qfj/QFJCodecSettings.java [hunk 3]
```diff
@@ -1,7 +1,24 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 package com.exactpro.th2.codec.qfj;
 
 import com.exactpro.th2.codec.api.IPipelineCodecSettings;
+import com.google.auto.service.AutoService;
 
+@AutoService(IPipelineCodecSettings.class)
```
Comment (cordwelt) [inline src/main/java/com/exactpro/th2/codec/qfj/QFJCodecSettings.java:21]: Redundant - we only we need factory to be discoverable

**Review by cordwelt — CHANGES_REQUESTED** [review]

**Review by Dmitriy-Yugay — APPROVED** [review]

File: src/main/java/com/exactpro/th2/codec/qfj/QFJCodec.java [hunk 4]
```diff
@@ -216,8 +232,168 @@ private void setFields(Map<String, Value> fieldsMap, quickfix.FieldMap qfjFieldM
         return groups;
     }
 
+    private Field<?> getField(int tag, String valueName, DataDictionary dataDictionary) {
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/java/com/exactpro/th2/codec/qfj/QFJCodec.java:215]: Could you please extract converter to a separate class

**Review by cordwelt — APPROVED** [review]

**Pull request merged**